docker-k3os-image-gen/docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+version: "3"
+services:
+  rpi-image-gen:
+    build: ../
+    environment:
+      TARGET: raspberrypi
+    volumes:
+      # bind ../config to our working dir config
+      # this ONLY works with unnamed volumes like below!!
+      - ../config:/usr/src/app/config
+      # result is in this named volume
+      - "images:/usr/src/app/images"
+      # this is the downloads cache
+      - "k3os-fw:/usr/src/app/deps"
```

Comment:
Having the downloads cache in a named volume is fine by me, but why is there a difference in how you bind the `config` directory and how you bind the `images` directory?

Reply:
Wouldn't it be easy & acceptable if we just bind the entire repository into the Docker container?

Reply:
the `config` directory is an input for the docker container; the way I do it here is the only way I know how the config directory is read every time, even if the images for the container already exist. The `images` and `cache` directory are output directories and need to persist, therefore a chooser named volumes and since under OSX and Windows they are actually in a VM we need to bind mount them ... Which also answers the question why docker-compose is in a different directory: since depending on which platform it runs (OSX, Windows or Linux) we may need different docker-compose ...